### PR TITLE
docs: modify docs on doc testing to reflect code snippet execution

### PR DESCRIPTION
**Pull request by magurotuna**
We landed "true" doc testing in https://github.com/denoland/deno/pull/25220. With this change, `deno test --doc` will actually _run_ code snippets written in JSDoc or markdown files, as well as type checking. This commit updates the corresponding doc pages to follow this change.

Preview of the updated pages:

- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/fundamentals/testing/#documentation-tests
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/fundamentals/typescript/#type-checking
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/reference/cli/check/
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/reference/cli/test/

### CODE REVIEW

File: runtime/fundamentals/testing.md [hunk 1]
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
...
+:::
 
-### Importing code in examples
+### Exported items are automatically imported
 
-You can import code from the surrounding module into the example code block by
-using the `import` keyword followed by the relative path to the module. This
-allows you to test the code in the example block in the context of the module
-being documented.
+Looking at the generated test code above carefully, you will notice that it
```
Comment (thisisjofrank) [inline runtime/fundamentals/testing.md:402]:
```suggestion
Looking at the generated test code above, you will notice that it
```

File: runtime/fundamentals/testing.md [hunk 2]
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
...
+includes the `import` statement to import the `add` fucntion even though the
```
Comment (thisisjofrank) [inline runtime/fundamentals/testing.md:403]:
```suggestion
includes the `import` statement to import the `add` function even though the
```

File: runtime/fundamentals/testing.md [hunk 3]
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
...
+original code block does not have it. Briefly, all items imported from the
+module being documented are automatically imported into the generated test code
+with the same name.
```
Comment (thisisjofrank) [inline runtime/fundamentals/testing.md:406]:
```suggestion
original code block does not have it. When documenting a module, any imported items are automatically included in the generated test code using with the same name.
```

**Review by marvinhagemeister — APPROVED** [review]

File: runtime/fundamentals/testing.md (hunk 3 above)
Comment (magurotuna) [reply runtime/fundamentals/testing.md:406]: Oh I now realized that the original sentence was wrong. What I meant is "any exported items from the module will automatically get imported in the generated test code using the same name". I'll fix the sentence anyway following your suggestion 👍

Comment (magurotuna) [reply runtime/fundamentals/testing.md]: Fixed [610af29](https://github.com/denoland/docs/pull/860/commits/610af29aeb9a24a412ff07a14bbf7df4867c0dd6)

**Comment by magurotuna** [comment]
@thisisjofrank Thanks for valuable review comments! I addressed them, and I realized I had forgotten to mention how to skip code blocks. This is now added in [7505cf4](https://github.com/denoland/docs/pull/860/commits/7505cf467f2eecb0e900c137d7fb5dbb1ec55e15). Can you give a final review?

**Review by thisisjofrank — APPROVED** [review]
LGTM!

**Pull request merged**